### PR TITLE
fix: handle edge case when month == 12

### PR DESCRIPTION
**Pull request by tushar5526**
Related to #91

### CODE REVIEW

**Review by karntrehan — CHANGES_REQUESTED** [review]
Lets add test cases for the change. There are edge cases we could miss.

File: src/school/school.service.v2.ts [hunk 1]
```diff
@@ -246,7 +246,7 @@ export class SchoolServiceV2 extends SchoolService {
           .replace('%table_v2%', tables.assessment_visit_results_v2)
           .replace('%mentor_id%', mentor.id.toString())
           .replace('%start_time%', (moment().month(item.month).year(item.year).date(1).startOf('day').format('YYYY-MM-DD HH:mm:ss')))
-          .replace('%end_time%', (moment().month(item.month + 1).year(item.year).date(1).startOf('day').format('YYYY-MM-DD HH:mm:ss')))
+          .replace('%end_time%', (moment().month(item.month + 1).year(item.year + (item.month + 1 == 12 ? 1 : 0)).date(1).startOf('day').format('YYYY-MM-DD HH:mm:ss')))
```
Comment (chinmoy12c) [inline src/school/school.service.v2.ts:249]: Why does `month + 1 == 12` add 1? Wouldn't that make the last month of the same year?? Can we have a single line of comment to explain these [MAGIC NUMBERS](https://en.wikipedia.org/wiki/Magic_number_(programming)#:~:text=In%20computer%20programming%2C%20a%20magic,see%20List%20of%20file%20signatures).

Comment (ChakshuGautam) [reply src/school/school.service.v2.ts:249]: Also consider replacing this to `DateTime` class with methods for formatting. Not able to make sense of what is happening and the kind of format that this is coverting it too. Also makes it difficutlt to write unit tests for this right now.

**Review by ChakshuGautam — CHANGES_REQUESTED** [review]

File: src/school/school.service.v2.ts (hunk 1 above)
Comment (tushar5526) [reply src/school/school.service.v2.ts:249]: `moment` is used across the whole codeabse at a lot of places, will make a separate ticket to switch to Datetime.

Comment (tushar5526) [reply src/school/school.service.v2.ts:249]: > Why does `month + 1 == 12` add 1? Wouldn't that make the last month of the same year?? Can we have a single line of comment to explain these [MAGIC NUMBERS](https://en.wikipedia.org/wiki/Magic_number_(programming)#:~:text=In%20computer%20programming%2C%20a%20magic,see%20List%20of%20file%20signatures).

Sure will add a comments, basically months are 0 indexed in `moment` library.

**Comment by tushar5526** [comment]
hey @karntrehan @ChakshuGautam, adding tests to NL would require a bigger PR and a refactoring of modules. As of now, I have switched to using the `moment.js`'s `add` function to add months which will take care of edge cases.

**Review by karntrehan — APPROVED** [review]
